modules/workspace.nix

```diff
@@ -279,19 +348,40 @@ in
     # kde tools. We then run this using an autostart script, where this is
     # run only on the first login (unless overrideConfig is enabled),
     # granted all the commands succeed (until we change the settings again).
-    programs.plasma.startup.startupScript."apply_themes" = (lib.mkIf anyThemeSet {
-      text = ''
-        ${if cfg.workspace.lookAndFeel != null then "plasma-apply-lookandfeel -a ${cfg.workspace.lookAndFeel}" else ""}
-        ${if cfg.workspace.theme != null then "plasma-apply-desktoptheme ${cfg.workspace.theme}" else ""}
-        ${if (cfg.workspace.cursor != null && cfg.workspace.cursor.theme != null) then
-        "plasma-apply-cursortheme ${cfg.workspace.cursor.theme}" +
-        (if cfg.workspace.cursor.size != null then " --size ${builtins.toString cfg.workspace.cursor.size}" else "")
-        else ""}
-        ${if cfg.workspace.colorScheme != null then "plasma-apply-colorscheme ${cfg.workspace.colorScheme}" else ""}
-        ${if cfg.workspace.iconTheme != null then "${pkgs.kdePackages.plasma-workspace}/libexec/plasma-changeicons ${cfg.workspace.iconTheme}" else ""}
-      '';
-      priority = 1;
-    });
+    programs.plasma.startup.startupScript = {
+      "apply_themes" = (lib.mkIf anyThemeSet {
+        text = ''
+          ${if cfg.workspace.lookAndFeel != null then "plasma-apply-lookandfeel -a ${cfg.workspace.lookAndFeel}" else ""}
+          ${if cfg.workspace.theme != null then "plasma-apply-desktoptheme ${cfg.workspace.theme}" else ""}
+          ${if (cfg.workspace.cursor != null && cfg.workspace.cursor.theme != null) then
+          "plasma-apply-cursortheme ${cfg.workspace.cursor.theme}" +
+          (if cfg.workspace.cursor.size != null then " --size ${builtins.toString cfg.workspace.cursor.size}" else "")
+          else ""}
+          ${if cfg.workspace.colorScheme != null then "plasma-apply-colorscheme ${cfg.workspace.colorScheme}" else ""}
+          ${if cfg.workspace.iconTheme != null then "${pkgs.kdePackages.plasma-workspace}/libexec/plasma-changeicons ${cfg.workspace.iconTheme}" else ""}
+        '';
+        priority = 1;
+      });
+
+      # Creates a startup script to set the desktop folder settings defined
+      # in the options under "cfg.workspace.desktop.icons".
```

Comment:
I also don't think these comments are really needed since it's self-explanatory by the name of the script.

Reply:
Okay, then I’ll drop it (I tend to comment a lot in general but especially in nix as it’s still hard for me to read and understand).